dask/dataframe/core.py

```diff
@@ -2600,6 +2604,10 @@ def clip_lower(self, threshold):
     def clip_upper(self, threshold):
         return self.map_partitions(M.clip_upper, threshold=threshold)
 
+    @derived_from(pd.DataFrame)
+    def squeeze(self):
+        return self.map_partitions(M.squeeze)
```

Comment:
What happens if we have partitions with single rows here?  Will they get squeezed when they shouldn't be?

Reply:
Sorry I missed that in your previous comment! Yes single-row partitions get squeezed into scalars whereas multiple row partitions get squeezed into `Series`.
```python

In [26]:
df = pd.DataFrame({'test': [0, 2, 3]})
d = dd.from_pandas(df, npartitions=3)
d.squeeze().compute()

Out[26]: 
0                                         0
1    1    2
2    3
Name: test, dtype: int64
dtype: object
```
Perhaps we could resolve this by setting `axis=1`:
``` python
    @derived_from(pd.DataFrame)
    def squeeze(self, axis=1):
        return self.map_partitions(M.squeeze, axis=axis)
```
However for versions of `pandas` < 0.20.2 the `axis` parameter isn't supported. We could try checking the size of the partition and converting that to a `Series` if it is a single row for versions < 0.20.2, or simply make `squeeze` unavailable for earlier versions?

Reply:
I wonder if it would make sense to rewrite this logic manually?

```python
if len(self.columns) == 1:
    return self[self.columns[0]]
else:
    return self
```

I don't know if this handles all cases though.  It would be good to ensure that the tests cover all relevant cases.

Reply:
I think that may be the least 'fiddly' solution.
We could add a `squeeze()` function in `methods.py`, and apply it using `map_partitions` as a class method in `core.py`?

Reply:
Thanks, was unaware of that. Will have a go at it. Sorry this is such a slow process, still feel a little unfamiliar with how dask works, hopefully that will improve soon.

Reply:
No worries.  Thanks for pitching in!  I'm quite happy to answer questions like this to help people get started on developing the project.

Reply:
I've put a `NotImplementedError` for `axis=1` in the method, since I'm not really sure how to access the single row in the dask dataframe ( if I understand correctly, `iloc` is not implemented?), is that acceptable?

Reply:
This seems sensible to me.  It would also be good to add a test that the user receives an informative message when they supply axis=0.  I usually do this with the following pattern:

```python
with pytest.raises(NotImplementedError) as info:
    df.squeeze(axis=0)

assert 'some informative text' in str(info.value)
```

Where 'some informative text' is something that tells them that this doesn't work.